### PR TITLE
OSSindex integration + Security updates

### PR DESCRIPTION
**Pull request by mcompastie**
In this PR, we (i) integrate the oss-audit to determine CVE flawing the code of the projet and (ii) update the dependencies identified as vulnerable to their latest maintenance versions.

A contributor can now inspect the project to assess if security updates have to be enforced by executing `./gradlew audit [--info]`.

### CODE REVIEW

**Comment by fviale** [comment]
One comment in general, could you make the task build depends on audit?

File: build.gradle [hunk 1]
```diff
@@ -48,6 +50,14 @@ subprojects {
 
     compileJava.options.compilerArgs << '-Xlint:-options' // remove warning about bootstrap class path
 
+
+    audit {
+         failOnError = false
+         rateLimitAsError = false
+         cache = "/tmp/ossindex.cache"
```
Comment (fviale) [inline build.gradle]: careful this is not cross-platform.
You maybe can get the tmp directory from gradle or using a java property

**Review by fviale — COMMENTED** [review]
One comment about cross-platform cache path

File: build.gradle (hunk 1 above)
Comment (mcompastie) [reply build.gradle]: Comment addressed

**Review by fviale — APPROVED** [review]
LGTM

**Review by Mykhailenko — APPROVED** [review]

**Review by mboussaa — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -157,9 +167,9 @@ project(':programming-core') {
 
     dependencies {
         compile(
-                'org.eclipse.jetty:jetty-server:9.2.14.v20151106',
-                'org.eclipse.jetty:jetty-servlet:9.2.14.v20151106',
-                'org.eclipse.jetty:jetty-xml:9.2.14.v20151106',
+                'org.eclipse.jetty:jetty-server:9.2.29.v20191105',
+                'org.eclipse.jetty:jetty-servlet:9.2.29.v20191105',
+                'org.eclipse.jetty:jetty-xml:9.2.29.v20191105',
```
Comment (nebilbenmabrouk) [inline build.gradle:172]: For the sake of homogeneity, why don't you use the same version of jetty as in the scheduling project (which is '9.2.24.v20180105').

Comment (mcompastie) [reply build.gradle:172]: It's WIP. We should be addressing this soon.

**Pull request merged**